internal/fwserver/server_getmetadata.go

```diff
@@ -44,15 +42,6 @@ func (s *Server) GetMetadata(ctx context.Context, req *GetMetadataRequest, resp
 	resp.Resources = []ResourceMetadata{}
 	resp.ServerCapabilities = s.ServerCapabilities()
 
-	metadataReq := provider.MetadataRequest{}
```

Comment:
Should we call `s.ProviderTypeName()` to set `providerTypeName` in order to preserve original behaviour?

Reply:
I also wondered this, it felt weird to make the call here to populate a cache used somewhere else. If we do want to preserve this, we could maybe change the naming up:

`s.SetProviderTypeName()`

But I went for the "lazy-load" approach 😆

Reply:
Just a clarification, the overall behavior is still preserved as there already exists a test for this: https://github.com/hashicorp/terraform-plugin-framework/blob/f147a4e18b39cfbee34b1d95adb0b5cfd12ec0a8/internal/fwserver/server_getmetadata_test.go#L156

The implementation detail has just been moved closer to where it's needed for the other RPCs to consume as well 👍🏻 

An alternative approach could be to set `s.ProviderTypeName()` at the top of every RPC